Makefile

```diff
@@ -77,14 +77,22 @@ manifests: controller-gen ## Generate manifests e.g. CRD, RBAC etc
 	$(Q)go generate ./crds/...
 
 # Static tests.
-.PHONY: test test-unit
+.PHONY: diff test test-unit verify verify-manifests verify-generate
+
+diff:
+	git diff --exit-code
 
 test: test-unit ## Run the tests
 
 TEST_PKGS:=$(shell go list ./...)
 test-unit: ## Run the unit tests
 	$(Q)go test -count=1 -short ${TEST_PKGS}
 
+
+verify-manifests: manifests diff
+verify-generate: generate diff
```

Comment:
I can't find any promise that GNU Make will build prerequisites in any particular relative order (other than before the target, of course).

Reply:
You could also just have the `verify` target be a driver that calls the verify-* commands in order instead of supplying them as recipes/dependencies.

Reply:
I was unable to find a guarantee but running the following test always resulted in the expected order:
```
makefile-test: mftest1 mftest2
mftest1:
	echo "1"

mftest2:
	echo "2"
```

Reply:
@benluddy pointed out that the prerequisites for the target above can be executed if the number of jobs is greater than 1.

Example command:
```
$ make makefile-test -j8
echo "1"
echo "2"
2
1
```